### DISCUSSION TITLE
Ignore sly specific stack frames.

### DISCUSSION THREAD

**Discussion by jgarte** [joaotavora/sly#620] [Q&A]
Is there a way to ignore the stack frames that belong to sly or slime code?

I just want to know if it is possible to remove or filter out those stack frames since they are noise to my debugging experience.

**Comment by joaotavora** [comment]
No, there isn't.  But this could be an interesting feature. 

Though not easy to define.  Let's take a look at a simple `(error "bla")` input from the REPL:

```
Backtrace:
 0: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ERROR "bla") #<NULL-LEXENV>)
 1: (EVAL (ERROR "bla"))
 2: ((LAMBDA NIL :IN SLYNK-MREPL::MREPL-EVAL-1))
 3: (SLYNK::CALL-WITH-RETRY-RESTART "Retry SLY mREPL evaluation request." #<FUNCTION (LAMBDA NIL :IN SLYNK-MREPL::MREPL-EVAL-1) {1001C9474B}>)
 4: ((LAMBDA NIL :IN SLYNK-MREPL::MREPL-EVAL-1))
 5: ((LAMBDA NIL :IN SLYNK::CALL-WITH-LISTENER))
 6: (SLYNK::CALL-WITH-BINDINGS ((*PACKAGE* . #<PACKAGE "COMMON-LISP-USER">) (*DEFAULT-PATHNAME-DEFAULTS* . #P"/home/capitaomorte/") (*) (**) (***) (/) ...) #<FUNCTION (LAMBDA NIL :IN SLYNK::CALL-WITH-LIST..
 7: (SLYNK-MREPL::MREPL-EVAL-1 #<SLYNK-MREPL::MREPL mrepl-1-1> "(error \"bla\")")
 8: (SLYNK-MREPL::MREPL-EVAL #<SLYNK-MREPL::MREPL mrepl-1-1> "(error \"bla\")")
 9: (SLYNK:PROCESS-REQUESTS NIL)
10: ((LAMBDA NIL :IN SLYNK::SPAWN-CHANNEL-THREAD))
11: ((LAMBDA NIL :IN SLYNK::SPAWN-CHANNEL-THREAD))
12: (SLYNK-SBCL::CALL-WITH-BREAK-HOOK #<FUNCTION SLYNK:SLYNK-DEBUGGER-HOOK> #<FUNCTION (LAMBDA NIL :IN SLYNK::SPAWN-CHANNEL-THREAD) {1001A67FDB}>)
13: ((FLET SLYNK-BACKEND:CALL-WITH-DEBUGGER-HOOK :IN "/home/capitaomorte/Source/Emacs/sly/slynk/backend/sbcl.lisp") #<FUNCTION SLYNK:SLYNK-DEBUGGER-HOOK> #<FUNCTION (LAMBDA NIL :IN SLYNK::SPAWN-CHANNEL-TH..
14: ((LAMBDA NIL :IN SLYNK::CALL-WITH-LISTENER))
15: (SLYNK::CALL-WITH-BINDINGS ((*PACKAGE* . #<PACKAGE "COMMON-LISP-USER">) (*DEFAULT-PATHNAME-DEFAULTS* . #P"/home/capitaomorte/") (*) (**) (***) (/) ...) #<FUNCTION (LAMBDA NIL :IN SLYNK::CALL-WITH-LIST..
16: ((LAMBDA NIL :IN SLYNK::SPAWN-CHANNEL-THREAD))
17: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
18: ((FLET "WITHOUT-INTERRUPTS-BODY-11" :IN SB-THREAD::RUN))
19: ((FLET SB-UNIX::BODY :IN SB-THREAD::RUN))
20: ((FLET "WITHOUT-INTERRUPTS-BODY-4" :IN SB-THREAD::RUN))
21: (SB-THREAD::RUN)
22: ("foreign function: call_into_lisp_")
23: ("foreign function: funcall1")
```

What frames would you like to ignore here? Which ones would you like to see?

Feel free to post a backtrace from one of your actual errors

**Reply by jgarte** [reply]
In the below screenshot of `linked-list.lisp` I would like to not see all the frames that mention slynk in the name:

![image](https://github.com/joaotavora/sly/assets/47760695/3b3230e4-66ef-4d06-9667-84dc5f065090)

I would like to see stack frame `3:` in this example.

I'll try to post a better example where I have 3 or more stack frames in `linked-list.lisp` that I do want to see. I had one like that the other day.

Let me know if you get the idea of what I'm after from this description.

**Reply by joaotavora** [reply]
I get _an_ idea, but not necessarily _the_ idea.  Please annotate your screenshot visually with the stackframes that you would like to see, and the ones that you would not like to see.  Better yet, use at test block
```
like this
```

**Comment by phoe** [comment]
Aren't they hidden by default under a "more" button? At least in my slime, the backtrace like this:

![Zrzut ekranu z 2023-10-01 23-29-44](https://github.com/joaotavora/sly/assets/15045546/cde469a9-2f6b-435e-9bce-71ce62cfa472)

And only after clicking it expands:

![Zrzut ekranu z 2023-10-01 23-30-17](https://github.com/joaotavora/sly/assets/15045546/bdcd15ce-ca6b-4e87-91ce-629f6c40649a)

**Reply by joaotavora** [reply]
That's true in SLY as well... I guess the feature is already there then :-)  But can we be sure that none of the uninteresting frames pop up above the hidden part?

**Comment by jgarte** [comment]
Pycharm calls the feature I am describing here "step into my code":

> When debugging, you might want to concentrate on your own code and prevent debugger from stepping into library classes.

Reference:

https://www.jetbrains.com/help/pycharm/stepping-through-the-program.html#step-into-my-code

**Reply by jgarte** [reply]
That said, there might be instances where I might want to step into 3rd party library code but not slynk code.

In this case I'd like a configuration where slynk library code is hidden from my stack frame list/view.
